src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1483,6 +1519,17 @@ private enum SettingId : ushort
             MaxHeaderListSize = 0x6
         }
 
+        /// <summary>Specifies when the data written needs to be flushed.</summary>
+        private enum FlushTiming
+        {
+            /// <summary>No specific requirement.  This can be used when the caller knows another operation will soon force a flush.</summary>
+            Eventually,
+            /// <summary>The data needs to be flushed soon, but it's ok to wait briefly for pending writes to further populate the buffer.</summary>
+            AfterPendingWrites,
+            /// <summary>The data must be flushed immediately.</summary>
+            Now
```

Comment:
I'm not sure we really need this.

Yeah, the RFC says some stuff about "immediately", and in some other cases we certainly don't want to wait very long to send the frame. But the delay between AfterPendingWrites and Now should be very, very small. The routines that actually populate the write buffer are doing a minimal amount of work, basically just pushing data into the write buffer. And in the worst case, we still have the buffer limit that will force a flush.

So it's not clear to me that we really get much by forcing flushes immediately as opposed to just letting pending writers complete and then flushing.

Reply:
BTW, re the RFC wording on SETTINGS ACK and "immediately", I take this to mean that the settings need to take effect before any subsequent requests are processed, and the SETTINGS ACK needs to be sent before any frames that depend on the SETTINGS change (e.g. a change to max concurrent streams that allows new requests to be sent) are sent. Which I believe we do already, regardless of whether the buffer is actually flushed right away or not -- and if we don't, then flushing the buffer won't actually help here...

Reply:
I can change the value passed for settings if it's not desirable there, but I think it's still worth keeping for Now.  In particular, for cancellation/aborts/etc., I think there's value in paying the potentially extra frame in order to get the reset on the wire asap, and not delay it at all, even if it's for a short period.  The benefit may be small, but so too is the cost.